frontend/src/components/VisitRouter/index.js

```diff
@@ -38,13 +41,30 @@ const VisitRouter = observer(() => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
-  const handleSubmitVisit = () => {
+  const handleSubmitVisit = async () => {
     // TODO: logic protecting against concurrent queue entries
     // if existing visit we are coming from QueueTable, so update visit
-    if (existingVisit.id) {
-      participantStore.updateVisit()
-    } else {
-      participantStore.createVisit()
+    try {
+      let isValid = await validateVisitForm(existingVisit)
+      if (existingVisit.id) {
+        !isValid.length
+          ? participantStore.updateVisit()
+          : isValid.map(error =>
+              participantStore.setSnackbarState(
+                `Theres an error in the ${error.name} field.`
+              )
+            )
+      } else {
```

Comment:
change `isValid` to `validationErrors`

i would invert this logic a bit:

"if validationErrors.length 
           return map set snackbar state

if existingvisit.id
  update
else
 create
"

Reply:
Inverting the logic is not super important, its just a little repetitive. I do want you to change the variable name here at the least though

Reply:
Cool. Yeah I took care of both things. Thanks for catching that.